tests/functional/test_install_extras.py

```diff
@@ -152,24 +152,14 @@ def test_install_fails_if_extra_at_end(
         script.scratch_path / "requirements.txt",
         expect_error=True,
     )
-    assert "Extras after version" in result.stderr
+    assert "Invalid requirement: 'requires_simple_extra>=0.1[extra]'" in result.stderr
 
 
 @pytest.mark.parametrize(
     "specified_extra, requested_extra",
     [
         ("Hop_hOp-hoP", "Hop_hOp-hoP"),
-        pytest.param(
-            "Hop_hOp-hoP",
-            "hop-hop-hop",
-            marks=pytest.mark.xfail(
-                reason=(
-                    "matching a normalized extra request against an"
-                    "unnormalized extra in metadata requires PEP 685 support "
-                    "in packaging (see pypa/pip#11445)."
-                ),
-            ),
-        ),
+        ("Hop_hOp-hoP", "hop-hop-hop"),
         ("hop-hop-hop", "Hop_hOp-hoP"),
     ],
 )
```

Comment:
@uranusjr by chance would know why this test would fail in python<3.10 and succeed in python>=3.10?

Reply:
Is this based on top of https://github.com/pypa/pip/pull/12002?

Otherwise, I expect that it's likely to be https://github.com/pypa/pip/pull/12267#issuecomment-1709983834.

Reply:
> fail in python<3.10 and succeed in python>=3.10?

I think you meant `<=3.10` and `>3.10` here. That'd be due to the switch to the importlib.metadata backend. We should remove the `safe_extra` call in there.

Reply:
aah, thanks, I'm now spotting the TODOs with PEP 685-related things to remove

Reply:
Well... now `test_install_special_extra` passes with python `>=3.11` and fails with `<=3.10`. :man_shrugging: 

I hope someone who already went into the extras normalization rabbit hole can look into it.

Reply:
The 3.10/11 split is due to pkg_resources and importlib.metadata normalising extras differently. This is where we make the switch between the two.

Is it possible to split this part into a separate PR? I can maybe find some time to look into this in the coming week or so.

Reply:
> Is it possible to split this part into a separate PR?

I'm not sure how? This looks closely tied to the `packaging` upgrade. But feel free to push to my branch.

Reply:
One thought - could we not just mark the test as `xfail` on older Python versions? As I understand it, we're testing if we correctly handle normalisation of extras. However, the reason the test fails on older Python versions is because *setuptools* doesn't normalise correctly, not because pip doesn't. That seems to me to be a good case for xfail. We can remove the xfail when (if) setuptools implements correct normalisation.

Reply:
The problem is that `setuptools.DistInfoDistribution` makes the `extras` attribute a list of extras normalised by `safe_extra`. But pip has canonicalised the requested extra, and it can't re-constitute the non-canonical value that's in `extras` (nor should it). I don't think this is easily fixable without a fairly extensive rewrite of our extra handling (or a fix to pkg_resources to store normalised extra names, which, to be fair, isn't required by PEP 685).

Do we have an understanding of *why* this has been triggered by the `packaging` update? I don't actually see how `packaging` is involved here. The relevant commit seems to be d1c5af81edb2e, but is that necessary? The TODO being removed says it's to support PEP 685, but we don't *actually* need to do that right now, do we?

Reply:
I think had test failures after upgrading packaging. I attempted to solve them by removing the TODOs, which led to other failures. If we don't address this all at once I worry that we'd change the current behaviour to change it again a release or two later, which will make things quite confusing for users.